src/http.cc

```diff
@@ -2247,18 +2247,44 @@ copyOneHeaderFromClientsideRequestToUpstreamRequest(const HttpHeaderEntry *e, co
             hdr_out->addEntry(e->clone());
         break;
 
-    case Http::HdrType::IF_NONE_MATCH:
```

Comment:
I suggest that we change caller code to avoid adding an ETag-based If-None-Match header if If-None-Match is already `*`, regardless of the other complex conditions currently checked in this copyOneHeaderFromClientsideRequestToUpstreamRequest() clause. That would be the best way to avoid generation of syntactically invalid headers by Squid code IMO.

If that suggestion is rejected, but this code continues to delete certain IF_NONE_MATCH header entries from hdr_out, then please update `copyOneHeaderFromClientsideRequestToUpstreamRequest() adds client ones` comment in the calling function code to clarify that this function may, under certain complex conditions, also _delete_ ETag-based header just added by the caller.